src/hooks/useActiveProject.tsx

```diff
@@ -50,26 +59,34 @@ export const ActiveProjectContextProvider = ({
         return;
       }
 
-      // TODO: Save for previously-selected project
+      storeProjectId(selectedProject.id);
 
       setHookReturnValue({
         activeProject: selectedProject,
         activeSubjectId: selectedSubject.subjectId,
       });
     },
-    [projectsResult.data, useMeResult.data],
+    [projectsResult.data, useMeResult.data, storeProjectId],
   );
 
   useEffect(() => {
-    if (projectsResult.data?.length) {
-      // TODO: Load previously-selected project
-
-      const firstProjectId = projectsResult.data?.[0].id;
-      if (firstProjectId) {
-        setActiveProjectId(firstProjectId);
+    if (projectsResult.data?.length && !storedProjectIdResult.isLoading) {
```

Comment:
`!storedProjectIdResult.isLoading` doesn't necessarily mean that `storedProjectIdResult.data` is populated.  e.g. it could be that `storedProjectIdResult.error` is populated instead?  I think you have the code within this if statement defensive enough to handle `storedProjectIdResult.data` being undefined, but there's probably an edge case bug in here (e.g. first project fetch succeeded, but subsequent one failed).  Not worried about this atm, just pointing it out as another factor in the react-query approach

Reply:
Resolved by refactor